specification/storagetasks/resource-manager/Microsoft.StorageTasks/stable/2023-01-01/storagetasks.json

```diff
@@ -0,0 +1,903 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- List of storage tasks in the given resource group retrieved and returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageTasksListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageTasks/storageTasks/{storageTaskName}/storageTaskAssignments": {
+      "get": {
+        "tags": [
+          "StorageTasks"
+        ],
+        "operationId": "ListStorageTaskAssignmentIds",
+        "description": "Lists all the storage tasks available under the given resource group.",
+        "x-ms-examples": {
+          "ListStorageTasksByResourceGroup": {
+            "$ref": "./examples/storageTasksList/ListStorageTaskAssignmentIds.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/StorageTaskName"
+          },
+          {
+            "name": "$maxpagesize",
+            "in": "query",
+            "type": "string",
+            "description": "Optional, specifies the maximum number of storage task assignment Ids to be included in the list response."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- List of storage task assignment Ids retrieved and returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageTaskAssignmentsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageTasks/storageTasks/{storageTaskName}/storageTaskAssignmentsExecutionSummary": {
```

Comment:
>storageTaskAssignmentsExecutionSummary [](http://example.com/codeflow?start=136&length=38)

The "storageTaskAssignmentsExecutionSummary" is not a valid resource type. A valid resource type should be in plural form, like virtualMachines, storageTasks etc. Looks like you need to modify this to a POST Action here:
POST /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageTasks/storageTasks/{storageTaskName}/getstorageTaskAssignmentsExecutionSummary. Please notice the "get" prefix for Action Name, as a POST Action should always be a verb.

Reply:
This was POST earlier, because it has query parameters one of the automation check fails stating that POST cannot have query parameters, so I had to change this to GET.

How to get around this ?

Reply:
Use POST Body Payload to pass the input, instead of query parameters.

Reply:
I am not sure, How that would work. Just like Api version, we also have Query parameters and not a request Body

$filters = '$filter=startTime gt ''2023-06-27T21:48:04.2982632Z'''
$url = "subscriptions/0c8875c7-e423-4caa-827a-1f0350bd8dd3/resourceGroups/hichhabr-rg1/providers/Microsoft.Storage/storageTasks/deletehottask/storageTaskAssignmentsExecutionSummary?api-version=2023-01-01&$filters"
armclient.exe POST $url

Reply:
as per IM chat, make below changes

GET storageTasks/{taskname}/reports/default

GET storageTasks/{taskname}/reports

Reply:
as per IM discussion, commit is pushed with request body for POST

Reply:
As discussed offline, this needs to eb updated to Collection GET with ARM Resource model response.

Reply:
@visingla-ms - I have pushed the change with the example. kindly take a look at it